src/backend/distributed/commands/alter_table.c

```diff
@@ -0,0 +1,811 @@
+{
+	if (TableReferencing(relationId))
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because table has a foreign key")));
+	}
+}
+
+
+/*
+ * EnsureTableNotReferenced checks if the table is referenced by another
+ * table and errors if it is.
+ */
+void EnsureTableNotReferenced(Oid relationId)
+{
+	if (TableReferenced(relationId))
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because a foreign key references to table")));
+	}
+}
+
+
+/*
+ * EnsureTableNotForeign checks if the table is a foreign table and errors
+ * if it is.
+ */
+void EnsureTableNotForeign(Oid relationId)
+{
+	char relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_FOREIGN_TABLE)
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because it is a foreign table")));
+	}
+}
+
+
+/*
+ * EnsureTableNotPartition checks if the table is a partition of another
+ * table and errors if it is.
+ */
+void EnsureTableNotPartition(Oid relationId)
+{
+	if (PartitionTable(relationId))
+	{
+		Oid parentRelationId = PartitionParentOid(relationId);
+		char *parentRelationName = get_rel_name(parentRelationId);
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because table is a partition"),
+						errhint("the parent table is \"%s\"",
+								parentRelationName)));
+	}
+}
+
+
+void
+CreateDistributedTableLike(Oid relationId, Oid likeRelationId, char *distributionColumn, int shardCount, bool shardCountIsNull, char * colocateWith)
+{
+	Var *distributionKey = NULL;
+
+	if (distributionColumn)
+	{
+		Relation likeRelation = try_relation_open(likeRelationId, ExclusiveLock);
+		relation_close(likeRelation, NoLock);
+		distributionKey = BuildDistributionKeyFromColumnName(likeRelation,
+															 distributionColumn);
+	}
+	else
+	{
+		distributionKey = DistPartitionKey(likeRelationId);
+	}
+
+	if (colocateWith == NULL)
+	{
+		Var *originalDistributionKey = DistPartitionKey(likeRelationId);
+		if ((distributionColumn == NULL || originalDistributionKey->vartype == distributionKey->vartype) && shardCountIsNull)
+		{
+			char *likeRelationName = get_rel_name(likeRelationId);
+			Oid schemaId = get_rel_namespace(likeRelationId);
+			char *schemaName = get_namespace_name(schemaId);
+			colocateWith = quote_qualified_identifier(schemaName, likeRelationName);
+		}
+		else
+		{
+			colocateWith = "default";
+		}
+	}
+	char partitionMethod = PartitionMethod(likeRelationId);
+	CreateDistributedTable(relationId, distributionKey, partitionMethod, shardCount, colocateWith, false);
+}
+
+
+void
+CreateCitusTableLike(Oid relationId, Oid likeRelationId, int shardCount)
+{
+	if (IsCitusTableType(likeRelationId, DISTRIBUTED_TABLE))
+	{
+		CreateDistributedTableLike(relationId, likeRelationId, NULL, shardCount, true, NULL);
+	}
+	else if (IsCitusTableType(likeRelationId, REFERENCE_TABLE))
+	{
+		CreateDistributedTable(relationId, NULL, DISTRIBUTE_BY_NONE, ShardCount, NULL, false);
```

Comment:
I guess it's fine, but the fact that we're using `ShardCount`  here and `shardCount`  just above makes this seem like a bug.